### PR TITLE
Issue #565: `__get_inferred_counterpart_for_fetching_of_branch()` now correctly infers local branch's remote counterpart for fetching of branch when the branch is associated with more than one remote

### LINKED ISSUE #565: Subtask of #562: `__get_inferred_counterpart_for_fetching_of_branch` -> returns only one local branch remote counterpart, the first one that meets the requirements from alphabetically ordered remote-branch pairs
Subtask of #562

### PR DESCRIPTION
**Pull request by amalota**

### CODE REVIEW

File: RELEASE_NOTES.md [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
 ## New in git-machete 3.11.2
 
+- fixed: `git machete` now correctly infers remote for fetching of branch when the branch is associated with more than one remote
+- fixed: `git machete` now correctly infers local branch's remote counterpart for fetching of branch when the branch is associated with more than one remote
```
Comment (PawelLipski) [inline RELEASE_NOTES.md]: These two are technically a different thing... but I'd just leave the first point; such level of detail isn't needed for release notes

**Comment by PawelLipski** [comment]
Pls note that since this branch is stacked on the top of another, you'lll have a very hard time untangling the history once the base branch is squashed-merged into `develop`... generally, there's a trilemma (choose 2 out of three):
1. true-merge (2-parent merge)
2. rebases and squashes
3. stacked PRs

Out of these 3, I select 2. and 3. (no merges, unless for hotfixes) to keep history linear at all times. In fact, git-machete generally is opinionated towards this choice.

**Comment by amalota** [comment]
> Pls note that since this branch is stacked on the top of another, you'lll have a very hard time untangling the history once the base branch is squashed-merged into `develop`... generally, there's a trilemma (choose 2 out of three):
> 
> 1. true-merge (2-parent merge)
> 2. rebases and squashes
> 3. stacked PRs
> 
> Out of these 3, I select 2. and 3. (no merges, unless for hotfixes) to keep history linear at all times. In fact, git-machete generally is opinionated towards this choice.

Hmm, AFAICR i already made some merges

File: git_machete/git_operations.py [hunk 2]
```diff
@@ -459,10 +459,14 @@ def get_combined_remote_for_fetching_of_branch(self, branch: LocalBranchShortNam
...
     def __get_inferred_counterpart_for_fetching_of_branch(self, branch: LocalBranchShortName) -> Optional[RemoteBranchShortName]:
-        for remote in self.get_remotes():
-            if f"{remote}/{branch}" in self.get_remote_branches():
-                return RemoteBranchShortName.of(f"{remote}/{branch}")
-        return None
+        remotes_containing_branch: List[str] = self.__get_remotes_containing_branch(branch)
+        if len(remotes_containing_branch) > 1 or len(remotes_containing_branch) == 0:
+            debug(f'Can\'t infer local branch\'s remote counterpart for fetching of branch.\n'
+                  f'There are {len(remotes_containing_branch)} remotes: {",".join(remotes_containing_branch)} '
+                  f'associated with {branch} branch.')
```
Comment (PawelLipski) [inline git_machete/git_operations.py]:
```suggestion
                  f'containing branch {branch}.')
```

File: git_machete/git_operations.py [hunk 3]
```diff
@@ -459,10 +459,14 @@ def get_combined_remote_for_fetching_of_branch(self, branch: LocalBranchShortNam
...
 
     def __get_inferred_counterpart_for_fetching_of_branch(self, branch: LocalBranchShortName) -> Optional[RemoteBranchShortName]:
-        for remote in self.get_remotes():
-            if f"{remote}/{branch}" in self.get_remote_branches():
-                return RemoteBranchShortName.of(f"{remote}/{branch}")
-        return None
+        remotes_containing_branch: List[str] = self.__get_remotes_containing_branch(branch)
+        if len(remotes_containing_branch) > 1 or len(remotes_containing_branch) == 0:
+            debug(f'Can\'t infer local branch\'s remote counterpart for fetching of branch.\n'
+                  f'There are {len(remotes_containing_branch)} remotes: {",".join(remotes_containing_branch)} '
```
Comment (PawelLipski) [inline git_machete/git_operations.py]:
```suggestion
                  f'There are {len(remotes_containing_branch)} remotes: {", ".join(remotes_containing_branch)} '
```

**Review by PawelLipski — COMMENTED** [review]
Again, can it be undrafted/merged already?

**Comment by amalota** [comment]
> Again, can it be undrafted/merged already?

Not sure about merging because i think that there are no tests for this function yet

**Review by PawelLipski — APPROVED** [review]

**Review by PawelLipski — APPROVED** [review]

File: tests/test_status.py [hunk 4]
```diff
@@ -268,14 +268,14 @@ def test_inferring_counterpart_for_fetching_of_branch(self, mocker: Any) -> None
...
-  | 
+  |
   o-bar
-    | 
+    |
     o-foo (untracked)
-      | 
+      |
       o-snickers
-        |
```
Comment (PawelLipski) [inline tests/test_status.py]: Pls open an issue to get rid of these trailing spaces in the output of `def status`

**Pull request merged**